src/ProjectTemplates/Web.ProjectTemplates/content/BlazorWeb-CSharp/BlazorWeb-CSharp/Components/Routes.razor

```diff
@@ -0,0 +1,18 @@
+@using BlazorWeb_CSharp.Components.Layout
+
+@*#if (UseWebAssembly) -->
+<Router AppAssembly="@typeof(App).Assembly" AdditionalAssemblies="new[] { typeof(Client.Pages.Counter).Assembly }">
```

Comment:
I think there some value in trying to keep some of the constructs consistent with the existing Blazor templates (unless we really think they are problematic, like the weird Shared folder). With that in mind, I prefer that this component continue to be called App.razor, like it is in the existing templates, and then we come up with some other name for the top level component, like Host, AppHost, Root, Main, ...

Reply:
I did do it exactly how you suggested earlier, but found it felt really strange to have two components whose names both sound like "I'm the top-level thing". That is, `App`+`AppHost` or `App`+`Host` are both really odd pairings, as there's nothing to indicate which one is the real boss. It feels disordered.

The pair `App`+`Routes` has the dual benefit that:

 * It's obvious that `Routes` just deals with this one particular concern, and is subservient to `App`
 * We get to keep `App` as the name for the root component (as in, `MapRazorComponents<App>`) which will be more intuitive to our existing devs

Reply:
Also a third benefit: longer term I think we should consider adding a component called `DefaultRouter` or `StandardRouter` that simply means "the boilerplate router that everyone has and never has reason to change". Then:

 * We can remove the whole `Router` component from the default template entirely
 * It feels much more natural to just put `@rendermode=Server` etc on the `<DefaultRouter>` inside `App` rather than having to decide whether to go and add `@rendermode Server` inside a child component

The thing stopping us from doing that right now is that people have to localize the "not found" text. When we can remove the "not found" text entirely (https://github.com/dotnet/aspnetcore/issues/48983) we'll be free to do it.